File: WoofIOS/SwiftPackages/Data/DataSource/Sources/DataSourceFakes/Favorites/FavoritesDataSourceFactory.swift
import DataSource
import Swinject
import SwinjectAutoregistration

public final class FavoritesDataSourceFactory {
    private let dataSource: FakeFavoritesDataSource

    public init(_ container: Container) {
        dataSource = container~>
    }

    @discardableResult
    public func withAddFavoriteError() -> Self {
        dataSource.addFavoriteError = .unavailable("Could not add favorite")
        return self
    }
}

File: WoofAndroid/data/datasource/src/testFixtures/kotlin/com/perrystreet/woof/datasource/favorites/FavoritesDataSourceFactory.kt
package com.perrystreet.woof.datasource.favorites

import org.koin.test.KoinTest
import org.koin.test.inject

class FavoritesDataSourceFactory : KoinTest {
    private val dataSource: FakeFavoritesDataSource by inject()

    fun withAddFavoriteError() = apply {
        dataSource.addFavoriteError = IllegalStateException("Could not add favorite")
    }
}
